src/Uno.Extensions.Core.Generators/PropertySelectorGenerator.cs

```diff
@@ -31,17 +35,315 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
 			Debugger.Launch();
 		}
 #endif
+
+		/*
+		    assemblyNameProvider: The output is the assembly name of the current compilation.
+
+			syntaxTreesProvider: The output is all syntax trees that contain invocations like Method(..., x => x..., ...)
+
+			interestingMethodNames: The output is method names from compilation references that have PropertySelector parameter
+			filteredSyntaxTreesProvider: The output is all matching InvocationExpressionSyntax nodes given interestingMethodNames.
+			finalProvider: The output is `PropertySelectorCandidate`s that are semantics-aware. This only includes calls to methods from metadata.
+
+			interestingMethodNamesFromSource: The output is method names from current compilation sources that have PropertySelector parameter
+			filteredSyntaxTreesFromSource: The output is all matching InvocationExpressionSyntax nodes given interestingMethodNamesFromSource.
+			finalProviderFromSource: The output is `PropertySelectorCandidate`s that are semantics-aware. This only includes calls to methods from source.
+
+
+			    ┌──────────────────────┐
```

Comment:
In case it's hard to read this figure in GitHub:

![image](https://github.com/unoplatform/uno.extensions/assets/31348972/d0c8a82c-a57a-47bb-ac7f-ff8ff9d53a36)

Reply:
However, when running `interestingMethodNames`, we cache the AssemblyIdentity and the names in the CWT. So the first run of this step is expensive, but later runs are cached (by us, not the compiler)